### DISCUSSION TITLE
Pre-existing crack

### DISCUSSION THREAD

**Discussion by Nikhil23121998** [peridigm/peridigm#286] [Q&A]
Is it possible to run the pre-existing crack sample in peridigm tensile test example?

**Comment by oldninja** [comment]
I don't see why not. However, the tensile test is a quasi-static one, so not sure how this would work with the crack propagation.

**Reply by Nikhil23121998** [reply]
I am trying to run one sample with pre-existing crack but it is showing this error,

Allocating global tangent matrix...
  number of rows = 2814
  number of nonzeros = 5360004

Load step 1, initial time = 0, final time = 1, convergence criterion = 1
  iteration 1: residual = 3.65705e+40
Warning:  Belos linear solver aborted with Belos::CGIterateFailure.
  --switching nonlinear solver to damped Newton and deactivating preconditioner--
Warning:  Belos linear solver aborted with Belos::CGIterateFailure.

Error:  Belos linear solver failed to converge.

Error:  Aborting analysis.

**Reply by oldninja** [reply]
Does it need to be a quasi-static solver?

**Reply by Nikhil23121998** [reply]
How can I change my solver?

**Reply by oldninja** [reply]
https://github.com/peridigm/peridigm/blob/master/examples/disk_impact/disk_impact.yaml#L71-L75

**Reply by Nikhil23121998** [reply]
I have tried to change my solver but it is showing this error,

Total number of time steps 24

Explicit time integration [==12% Complete==================] /home/nikhil/peridigm/src/core/Peridigm.cpp:1552:

Terminate test that evaluated to true: !std::isfinite((*force)[i])

**** NaN returned by force evaluation.

**Reply by oldninja** [reply]
Any of the tests work properly?

**Reply by Nikhil23121998** [reply]
Test without crack is working properly.

**Reply by oldninja** [reply]
You may want to try switching it to Verlet and adding some mass scaling to see if it works...

**Reply by jritta** [reply]
> I am trying to run one sample with pre-existing crack but it is showing this error,
> 
> Allocating global tangent matrix... number of rows = 2814 number of nonzeros = 5360004
> 
> Load step 1, initial time = 0, final time = 1, convergence criterion = 1 iteration 1: residual = 3.65705e+40 Warning: Belos linear solver aborted with Belos::CGIterateFailure. --switching nonlinear solver to damped Newton and deactivating preconditioner-- Warning: Belos linear solver aborted with Belos::CGIterateFailure.
> 
> Error: Belos linear solver failed to converge.
> 
> Error: Aborting analysis.

Simulating quasi-static crack propagation with the implemented quasi-static solvers is always difficult. As the error messages says the solver could not converge and thus didn't find a solution.

**Reply by jritta** [reply]
> I have tried to change my solver but it is showing this error,
> 
> Total number of time steps 24
> 
> Explicit time integration [==12% Complete==================] /home/nikhil/peridigm/src/core/Peridigm.cpp:1552:
> 
> Terminate test that evaluated to true: !std::isfinite((*force)[i])
> 
> **** NaN returned by force evaluation.

Looks to me like an  instability in explicit time integration. Possible solutions are, as already suggested, density scaling or a reduction of the safety factor for the time step. The cause of the instability can have various reasons, maybe you provide a complete example so we can reproduce it. But as I mentioned before, simulating quasi-static crack propagation in Peridigm is very challenging in my experience...

**Comment by andrisfreimanis** [comment]
You can try to add the line "Belos Linear Solver: BlockGMRES" to your QS solver parameters and reduce the time step, but crack propagation with the QS solver will be difficult.

________________________________
